core/src/main/scala/org/apache/spark/package.scala

```diff
@@ -41,7 +41,53 @@ package org.apache
  * level interfaces. These are subject to changes or removal in minor releases.
  */
 
+import java.util.Properties
+
 package object spark {
-  // For package docs only
-  val SPARK_VERSION = "2.0.0-SNAPSHOT"
+
+  private object SparkBuildInfo {
+    val resourceStream = Thread.currentThread().getContextClassLoader.
+      getResourceAsStream("spark-version-info.properties")
+
+    val (
+      spark_version: String,
```

Comment:
We are using all caps because these are constants and the underscores to improve the readability.

More or less I have tried to be consistent following how SPARK_VERSION is named and used throughout.